### ISSUE TITLE
gce: how to specify service_account_permissions?

### ISSUE THREAD

**Issue by ottob** [ansible/ansible-modules-core#2295]
I've tried both formats below but the created instance only get storage-ro permission. I'm running ansible-2.0.0-0.3.beta1 and libcloud 0.18.0.

```
service_account_permissions:
  - computeaccounts-ro
  - logging-write
  - storage-rw
```

```
service_account_permissions: ["computeaccounts-ro", "logging-write", "storage-rw"]
```

No errors from ansible:

```
ok: [127.0.0.1] => (item={u'status': u'RUNNING', u'network': u'default', u'zone': u'europe-west1-d', u'tags': [u'www-tag'], u'image': u'backports-debian-7-wheezy-v20150929', u'disks': [u'www3'], u'name': u'www3', u'public_ip': u'x.x.x.x', u'private_ip': u'y.y.y.y', u'machine_type': u'f1-micro', u'metadata': {}})
```

Any idea what's going wrong @br0ziliy or @erjohnso?

**Comment by ottob** [comment]
Perhaps related to https://github.com/ansible/ansible-modules-core/issues/1888?

I also tried both formats with only `storage-rw`. But the created instance still only get storage-ro.

**Comment by robynbergeron** [comment]
@ottob -- thanks for submitting this bug report.

@erjohnso -- thanks in advance for taking a look at this bug report and for your ongoing work in maintaining this module.

**Comment by robynbergeron** [comment]
[module: cloud/google/gce.py]

**Comment by ottob** [comment]
```
$ export LIBCLOUD_DEBUG=/home/myuser/libcloud.log
$ export LIBCLOUD_DEBUG_PRETTY_PRINT_RESPONSE=1
$ ansible-playbook -i ansible_hosts site.yml
```

I tried this to see if something seemed wrong with the generated json. The instance is created (with incorrect permissions) but I don't get any libcloud.log file. Shouldn't this work?

Any clues to how to debug this further would be much appreciated.

**Comment by ottob** [comment]
Ping?

**Comment by supertom** [comment]
Hi @ottob

Short answer, yes, related to #1888.

I was not actually able to duplicate the exact issue, as when I run with the versions you specified (which, by now, of course are quite old), I receive a bad-permissions message involving computeaccounts-ro  and logging-write.

To compound the issue, adding more confusions for users, the "choices" listed in the [gce ansible module](https://github.com/ansible/ansible-modules-core/blob/d659c79db7335a7a81594f7d977cb6bb67e9c5a3/cloud/google/gce.py#L62) don't actually match the [scopes that were hardcoded](https://github.com/apache/libcloud/blob/664135b862eae7fb6c395c715db1cd61fa4c2540/libcloud/compute/drivers/gce.py#L973) in 0.18.0 of libcloud.

I recommend we close this issue as the root issue, handling scopes, is filed (#1888).  If you need this functionality today you could patch [libcloud](https://github.com/apache/libcloud/blob/664135b862eae7fb6c395c715db1cd61fa4c2540/libcloud/compute/drivers/gce.py#L973) with the scopes you need from [here](https://developers.google.com/identity/protocols/googlescopes#clouduseraccountsvm_alpha).

**Comment by ryansb** [comment]
Closing this, will be resolved with #1888
